glide-core/src/client/value_conversion.rs

```diff
@@ -11,6 +11,11 @@ pub(crate) enum ExpectedReturnType<'a> {
         key_type: &'a Option<ExpectedReturnType<'a>>,
         value_type: &'a Option<ExpectedReturnType<'a>>,
     },
+    // Second parameter is a function which returns true if value needs to be converted
+    SingleOrMultiNode(
+        &'a Option<ExpectedReturnType<'a>>,
+        Option<fn(Value) -> bool>,
```

Comment:
Not sure how difficult it would be to change, but generally you should prefer taking a generic implementing a closure trait like `F: Fn(Value) -> bool` if possible because plain function pointers can't capture their environment like closures can. We don't really need the extra flexibility for our existing cases, so this isn't critical to do. Might not be worth making this change if it involves an extra `Box` alloc, but maybe worth a try at least.

Reply:
Do you mean something like this?
```rs
pub(crate) enum ExpectedReturnType<'a, F: Fn(Value) -> bool> {
    Map {
        key_type: &'a Option<ExpectedReturnType<'a, F>>,
        value_type: &'a Option<ExpectedReturnType<'a, F>>,
    },
    // Second parameter is a function which returns true if value needs to be converted
    SingleOrMultiNode(
        &'a Option<ExpectedReturnType<'a, F>>,
        Option<F>,
    ),
...
```
(I also need to update all references to `ExpectedReturnType` type)
Or
```rs
pub(crate) enum ExpectedReturnType<'a> {
    Map {
        key_type: &'a Option<ExpectedReturnType<'a>>,
        value_type: &'a Option<ExpectedReturnType<'a>>,
    },
    // Second parameter is a function which returns true if value needs to be converted
    SingleOrMultiNode(
        &'a Option<ExpectedReturnType<'a>>,
        Option<&'a dyn Fn(Value) -> bool>,
    ),
...
```

Reply:
I was thinking something more like the latter. I was actually expecting we'd take `Box<dyn fn(Value) -> bool` so we could own the closure, but we could probably get away with the reference instead. I'm aware this uses dynamic dispatch, but I think the performance hit is fairly negligible here.